src/KDSoapClient/KDSoapClientInterface.h

```diff
@@ -346,6 +346,26 @@ class KDSOAP_EXPORT KDSoapClientInterface
       */
     void setTimeout(int msecs);
 
+    /**
+     * @brief setUseWSAddressing
+     * @param useWSAddressing
+     * If this parameter is set to true then the outgoing messages SOAP action parameter
+     * will be added as an Action field of the SOAP envelope's header
+     * (instead of appending it to the Content-type HTTP header).
+     * This will make the KDSOAP generated SOAP requests compatible with the WS-Addressing
+     * specification. For a request body example please see:
+     * https://www.w3.org/Submission/ws-addressing/#_Toc77464324
+     */
+    void setUseWSAddressing(bool useWSAddressing = true);
+
+    /**
+     * @brief useWSAddressing
+     * @return
+     * Returns true if the actions of the sent messages are passed in the SOAP envelope's
+     * header section instead of sending them in the request's URL parameters.
+     */
+    bool useWSAddressing() const;
```

Comment:
Again personally I would make it useWSAddressingAction, as it only affects the action parameter. The other WS-Addressing parameters are not used with this option. But I let others decide whether that is important.

Reply:
Ah, the usual issue. Naming a method after what it does, or after when it should be used. The latter might make sense when one wants to hide some details about what it does, but in general that's a slippery slope... like here.

After reading https://docs.microsoft.com/en-us/dotnet/framework/wcf/feature-details/messaging-protocols I believe this should indeed be renamed, because it *is* possible to use the action HTTP parameter even when using WS-Addressing, as long as it matches the action specified in the XML -- it's an optimization so that message dispatch doesn't need to parse the XML.

Therefore the logic here should not be "if you're using WSAddressing, tell us and we'll skip the action parameter".
It should simply be "Do you want the action parameter emitted? Say no if your server is broken or if you're too lazy to set it to the right value."

The method name shouldn't even refer to WSAddressing, the action parameter exists independently from that (on the contrary, it mostly makes sense when WSAddressing is NOT used).

Therefore I suggest something like `setSendHTTPActionParameter(bool)`. With an explanation in the documentation that matches the one on the above webpage [unless you find contradictory information elsewhere....].

(The trick is to not confuse SOAP-1.1 users, this does not affect the SOAPAction HTTP header).
setSendSoap12HTTPContentTypeActionParameter becomes a bit long :-)

Reply:
In the WSDL what the server provides I have a
`<wsaw:UsingAddressing/>` tag while the linked docs mentions the `<wsap:UsingAddressing />` for turning on the WS-Addressing 2004/08.

In my case the kdwsdl2cpp generated code sent the correct action in the HTTP parameter what is in the WSDL's soap12:address field location attribute, so the R2221 condition of the mentioned spec was satisfied before this PR too:

Error message while action is in the header:

```
"Fault 299: Error transferring ... - server replied: Cannot process the message because the content 
type 'application/soap+xml;charset=utf-8;action=http://localhost/nd/communication/IMasterData
/GetProductsSince' was not the expected type 'application/soap+xml; charset=utf-8'
```

Relevant WSDL part:

```
<wsdl:operation name="GetProductsSince">
      <soap12:operation soapAction="http://localhost/nd/communication/IMasterData/GetProductsSince" style="document"/>
      <wsdl:input>
        <soap12:body use="literal"/>
      </wsdl:input>
      <wsdl:output>
        <soap12:body use="literal"/>
      </wsdl:output>
      <wsdl:fault name="NDFaultContractFault">
        <soap12:fault name="NDFaultContractFault" use="literal"/>
      </wsdl:fault>
    </wsdl:operation>
```

And I got the same fault even if the request's wsa:Action is correct, so my server implementation seems to be ignoring R2222 as well:

```
<?xml version="1.0" encoding="UTF-8"?><soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope" xmlns:soap-enc="http://www.w3.org/2003/05/soap-encoding" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:wsa="http://www.w3.org/2005/08/addressing" xmlns:n1="http://localhost/nd/communication/"><soap:Header><wsa:Action>http://localhost/nd/communication/IMasterData/GetProductsSince</wsa:Action></soap:Header><soap:Body><n1:GetProductsSince><n1:shopId>a529a279-dc92-4943-93b8-d2912941381b</n1:shopId><n1:modifiedSince>2019-11-21T10:59:22.218</n1:modifiedSince></n1:GetProductsSince></soap:Body></soap:Envelope>
```